semantic_conventions/trace/tls.yaml

```diff
@@ -0,0 +1,204 @@
+groups:
+  - id: tls
+    prefix: tls
+    type: span
+    brief: "This document defines semantic conventions for TLS/SSL client and server Spans."
+    attributes:
+      - id: version
+        requirement_level: required
+        brief: >
+          Numeric part of the version parsed from the original string of the
+          negotiated [SSL/TLS protocol version](https://www.openssl.org/docs/man1.1.1/man3/SSL_get_version.html#RETURN-VALUES)
+        type: string
+        examples: ["1.2", "3"]
+      - id: version_protocol
```

Comment:
maybe `tls.protocol_name` ?

Reply:
Yeah, I agree, let me fix this.

Reply:
applied

Reply:
> actually, to make it consistent with `network.protocol.*`, should it be `tls.protocol.name`?
> 
> I understand that's not how ECS defines it, but suggest to stay consistent with other otel definitions

@lmolkova 
I'd say this case is not one-to-one comparable with `network.protocol.*` fields. 

The reasons are:

- The `SSL/TLS` strings are often considered as a part of the version. [Here is a blogpost explaining this](https://kinsta.com/knowledgebase/tls-vs-ssl/). Especially, `Well, TLS is actually just a more recent version of SSL` in that blog post. 
- So, taking that into account ssl/tls semantically refer to the same type of protocol. 
- The `protocol.name` is here already clear from the namespace `tls.*`
- So, the reasoning for `tls.version` and `tls.version_protocol` is that a full version string is something like `TLSv1.3`, `SSLv2`, etc. and this version is being split into the numeric part `3` --> `tls.version` and the protocol part `TLS` --> `tls.version_protocol`

Don't get me wrong, `tls.protocol.name` and `tls.protocol.version` wouldn't be wrong and I'm not against that proposal per se. **_BUT_**, IMHO the argument of consistency with `network.protocol.*` in that case is very weak (given the explanation above) compared  to the fact that we would break ECS users  (without need, since there are no comparable attributes in semconv and we are not doing a trade-off between breaking ECS users vs. breaking OTel users in this case).

Reply:
@AlexanderWert thanks for the explanation! 
So ECS would still have two attributes: one for name and one for version. 
The consistency is indeed not a super-strong argument, but I find `tls.version` and `tls.version_protocol` combination in ECS difficult to understand without reading the docs. 

it seems that if we designed it from the scratch in otel, we'd start with `tls.protocol.name` (tls, SSL, dtls , mtls, future things) and `tls.protocol.version`.
And from my side, there seem to be a trade-off between clear naming (and consistency) and backward-compatibility with ECS

Reply:
resolving since outdated